tests/testthat/test-renderer1-geom-abline.R

```diff
@@ -75,3 +72,56 @@ abline.at.5 <- slope*(cxy$cx-abl$x1)+abl$y1
 test_that("abline with negative slope intersects point", {
   expect_equal(abline.at.5, cxy$cy)
 })
+
+# Test for testing that geom_ablines are clipped to the plot area even after update_axes is called
+data(mtcars)
+# Grouping variable
+mtcars$cyl <- as.factor(mtcars$cyl)
+cyl.levels <- levels(mtcars$cyl)
+# Generate 60 ablines (20 per cyl), all visible at once
+set.seed(123)
+abline_data <- do.call(rbind, lapply(cyl.levels, function(cyl_val) {
+  data.frame(
+    slope = runif(20, -2, 2),
+    intercept = runif(20, 0, 500),
+    cyl = cyl_val
+  )
+}))
+viz <- list(
+  title = "geom ablines clipped to plot area",
+  allablines = ggplot() +
+  theme_animint(update_axes = c("x", "y"), height=400, width=400) +
+    geom_point(aes(mpg, disp, color = cyl), data = mtcars, showSelected = "cyl") +
+    geom_abline(
+      aes(slope = slope, intercept = intercept, color = cyl),
+      size = 1,
+      data = abline_data
+    ) +
+    ggtitle("All ablines, all cyl groups together"),
+    selector.types = list(cyl = "single")
+)
+info <- animint2HTML(viz)
+ablines <- getNodeSet(info$html, '//svg//g[contains(@class, "geom2_abline_allablines")]//line')
+
+get_line_coords <- function(line) {
+  sapply(c("x1", "x2", "y1", "y2"), function(a) as.numeric(xmlGetAttr(line, a)))
+}
+
+test_that("visible ablines are identified correctly after update_axes", {
+  visible_lines <<- ablines[sapply(ablines, function(line) {
+    coords <- get_line_coords(line)
+    (coords["x1"] != coords["x2"] || coords["y1"] != coords["y2"]) &&
+      all(coords >= 0 & coords <= 400)
+  })]
+  cat(sprintf("\nVisible lines: %d/%d\n", length(visible_lines), length(ablines)))
+  expect_gt(length(visible_lines), 0)
+})
+
+test_that("visible lines are properly clipped within plot area", {
+  if (length(visible_lines) > 0) {
+    coords <- t(sapply(visible_lines, get_line_coords))
+    expect_true(all(coords >= 0 & coords <= 400), info = "All visible line coordinates should be within the plot area")
+  } else {
+    skip("No visible lines found for clipping test")
```

Comment:
is this skip branch necessary because of the runif randomness?

could you please refactor to use a known number of visible lines, to avoid this if else?